File: user-service/src/main/java/rs/raf/user_service/repository/ClientRepository.java
package rs.raf.user_service.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import rs.raf.user_service.entity.BaseUser;
import rs.raf.user_service.entity.Client;

import java.util.Optional;

public interface ClientRepository extends JpaRepository<Client, Long> {
    Optional<Client> findByEmail(String email);
    Optional<Client> findByJmbg(String jmbg);

}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/repository/ClientRepository.kt
package pack.userservicekotlin.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.stereotype.Repository
import pack.userservicekotlin.domain.entities.Client
import java.util.*

@Repository
interface ClientRepository :
    JpaRepository<Client, Long>,
    JpaSpecificationExecutor<Client> {
    fun findByEmail(email: String): Optional<Client>

    fun findByJmbg(jmbg: String): Optional<Client>
}
